scripts/intree-extensions.ts

```diff
@@ -0,0 +1,94 @@
+import path from "path"
+import fse from "fs-extra"
+import execa from "execa"
+import { Writable } from "stream"
+
+function firstLongest(a: string, b: string) {
+  if (a.length >= b.length) {
+    return a
+  }
+
+  return b
+}
+
+function makeSink(prefix: string) {
+  let buf = ""
+
+  return new Writable({
+    write(chunk: any, encoding: string, cb: () => void) {
+      buf += chunk
+
+      const parts = buf.split(/\r?\n/)
+      const endsWithNL = buf.endsWith("\n")
+      if (endsWithNL) {
+        buf = parts.pop()
+      } else {
+        buf = ""
+      }
+
+      for (const line of parts) {
+        console.log(`${prefix}${line}`)
+      }
+
+      cb()
+    }
+  })
+}
+
+const args = process.argv.slice(2) // remove node call and file name call
+
+async function buildInTreeExtension(extensionPath: string, prefix: string) {
+  console.log(`Building: ${extensionPath}`)
+
+  const install = execa("yarn", { cwd: extensionPath })
+  install.stdout.pipe(makeSink(prefix))
+  await install
+
+  const build = execa("yarn", ["run", "build"], { cwd: extensionPath })
+  build.stdout.pipe(makeSink(prefix))
+  await build
+}
+
+async function testInTreeExtension(extensionPath: string, prefix: string) {
+  console.log(`Testing: ${extensionPath}`)
+
+  const test = execa("yarn", ["test"], { cwd: extensionPath })
+  test.stdout.pipe(makeSink(prefix))
+  await test
+}
+
+async function main() {
+  const cwd = process.cwd()
+  const [extensionsFolder, action] = args
```

Comment:
should you test if these are empty? (user could run `yarn intree-extensions` manually?)

Reply:
Probably, good catch.